### DISCUSSION TITLE
Is it possible to use EfficientDet Lite on ARTPEC8?

### DISCUSSION THREAD

**Discussion by garaujo23** [AxisCommunications/acap-computer-vision-sdk-examples#38] [Q&A] [answered]
I have run the same object detector python and got it to work with the included model. I then tried to use the efficientdet lite 0 model from [here](https://coral.ai/models/object-detection/).  Using the CPU model I get timeout errors and only one detection showing Time for call to inference-server: 1602 ms

**Comment by Corallo** [comment, marked as answer]
Hello @garaujo23 

We don't recommend to use that model if the speed is relevant on the application. EfficientDet has some characteristics that don't pair well with Artpec8, so it is normal to have inference time > 1s.
You will get much better performances with SSD-mobilenet.

**Comment by garaujo23** [comment]
Thanks, will try SSDlite

**Comment by garaujo23** [comment]
@Corallo So have spent a few days trying to retrain SSD-mobilenet v1 and v2 and having some issues with smaller objects. Seems it is a [common issue](https://github.com/tensorflow/models/issues/3196) with mobilenet. What is the most efficient way to get a cropped section of an image using the videosdk? Assume normal opencv operations will be slow.

**Reply by Corallo** [reply]
The vdo API doesn't offer a way to crop the videostream, it is only possible to change its resolution.
Cropping shouldn't be an expensive operation, in python you can just use [slicing](https://numpy.org/doc/stable/user/basics.indexing.html#slicing-and-striding) to get the cropped part of the image that you require. Slicing in python creates a [view](https://numpy.org/doc/stable/user/basics.copies.html#view) of the original image, which doesn't involve the creation of a copy of it, and should therefore be fast.

**Reply by garaujo23** [reply]
Am trying to test a smaller crop model, in testing I get the following:
status = StatusCode.DEADLINE_EXCEEDED
object-detector-python_1  |     details = "Deadline Exceeded"

Is there a way to set the deadline termination from the larod server?

**Reply by Corallo** [reply]
Hello, that deadline is not from the larod server, but from the grpc request and it is set to 3 seconds.
At the moment, there isn't a "simple way" to change it.

The timeout is set [here](https://github.com/AxisCommunications/acap-computer-vision-sdk/blob/6fd2c8a277a2123c6c7d35b1de426842cefecc6a/sdk/tfserving/tf_proto_utils.py#L136) in the SDK.

Instead of changing it in the SDK and rebuild it (which will take several hours), copy that file, modify the timeout, and replace it in your app in /usr/lib/python3.8/site-packages/tf_proto_utils.py

However, 3 seconds for the inference should be enough.
If you are encountering this timeout the problem might be elsewhere,  check the following before trying to increase the timeout:

1) Try to wait up to 1 pr 2 min for the first inference. During the first inference the model gets converted to an internal format, this conversion normally takes a few hundreds of milliseconds, but sometimes for complex models could take more.
2) More commonly, it can happen that the inference server stops because of an error. (typically can't load the model). In this case, the app keeps sending messages to the inference server, resulting in just many timeouts. Look at the log and eventually at docker -H <ip_camera> ps to make sure that the inference server is still running.
